tests/test_typing.py

```diff
@@ -0,0 +1,182 @@
+from typing import Any, Dict, List, Optional
```

Comment:
This can probably go to test_base_config.py

Reply:
Done in https://github.com/omry/omegaconf/pull/749/commits/7f5892e45897c65ca2b8fcda876e83b792dffd67.